app/app.go

```diff
@@ -8,6 +8,16 @@ import (
 	"net/http"
 )
 
+type App struct {
+	Srv *Server
```

Comment:
In turn, an instance of `Server` contains a `Store`, a `WebSocketRouter`, etc. Why do we need the extra layer of indirection that is provided by `App`? 

Why not change the methods to have a `Server` receiver instead of an `App` receiver? i.e., exactly what you did, but using `Server` instead of `App` to avoid introducing another layer of indirection?